app/ViewDeals.tsx

```diff
@@ -0,0 +1,113 @@
+import { ReactElement, useEffect, useState } from "react";
+import { Wrapper } from "./ui/Wrapper";
+import { styled } from "styled-components";
+import { Deal, DealAndStore } from "./types";
+import {
+  darkGrey,
+  black,
+  greyInactiveButton,
+  skyBlue,
+} from "./ui/colourLibrary";
+import { LargeButton } from "./ui/LargeButton";
+import { getDeals } from "@/server/getDeals";
+import { LoadingSpinner } from "./ui/LoadingSpinner";
+
+export function ViewDeals(props: { onCreateClick: () => void }): ReactElement {
+  const { onCreateClick } = props;
+  const [dealList, setDealList] = useState<DealAndStore[] | undefined>(
+    undefined
+  );
+  const [dealBeingUpdated, setDealBeingUpdated] = useState<Deal | undefined>();
+  useEffect(() => {
+    const fetchData = async () => {
+      const result = await getDeals();
+      setDealList(result);
+    };
+    fetchData();
+  }, []);
+
+  // if (dealBeingUpdated !== undefined) {
+  //   return (
+  //     <Wrapper>
+  //    <UpdateDeal
+  //         onBackClick={() => {
+  //           setDealBeingUpdated(undefined);
+  //         }}
+  //         onUpdateDealSuccess={async () => {
+  //           const result = await getDeals();
+  //           setDealList(result);
+  //           setDealBeingUpdated(undefined);
+  //         }}
+  //         deal={dealBeingUpdated}
+  //       ></UpdateDeal>
+  //     </Wrapper>
+  //   );
+  // }
```

Comment:
bad, ordinarily you should not leave this junk in but we'll let it slide this time

Reply:
https://github.com/auberginearms/groceryDatabaseApp/pull/26/commits/cacf23d8e39caeac6f435633507aa6b498f47cf6

Reply:
https://github.com/auberginearms/groceryDatabaseApp/pull/26/commits/53142294ecd6b2b814512951ecdbcb0b4bd30232